src/Uno.UI/UI/Xaml/UIElement.cs

```diff
@@ -215,6 +216,91 @@ internal static Matrix3x2 GetTransform(UIElement from, UIElement to)
 			return matrix;
 		}
 
+		internal Thickness GetPadding()
+		{
+			var property = GetPaddingProperty();
+			return property == null ? Thickness.Empty : (Thickness)GetValue(property);
+		}
+
+		internal Thickness GetBorderThickness()
+		{
+			var property = GetBorderThicknessProperty();
+			return property == null ? Thickness.Empty : (Thickness)GetValue(property);
+		}
+
+		internal DependencyProperty GetPaddingProperty()
+		{
+			switch (this)
+			{
+				case Panel _:
+					return Panel.PaddingProperty;
+
+				case Control _:
+					return Control.PaddingProperty;
+
+				case ContentPresenter _:
+					return ContentPresenter.PaddingProperty;
+
+				case Border _:
+					return Border.PaddingProperty;
+			}
+
+			return GetDependencyPropertyUsingReflection("PaddingProperty");
+		}
+
+		internal DependencyProperty GetBorderThicknessProperty()
+		{
+			switch (this)
+			{
+				case Panel _:
+					return Panel.BorderThicknessProperty;
+
+				case Control _:
+					return Control.BorderThicknessProperty;
+
+				case ContentPresenter _:
+					return ContentPresenter.BorderThicknessProperty;
+
+				case Border _:
+					return Border.BorderThicknessProperty;
+			}
+
+			return GetDependencyPropertyUsingReflection("BorderThicknessProperty");
+		}
+
+		private static readonly Dictionary<(Type type, string property), DependencyProperty> _dependencyPropertyReflectionCache =
+			new Dictionary<(Type, string), DependencyProperty>();
+
+		private DependencyProperty GetDependencyPropertyUsingReflection(string propertyName)
+		{
+			var type = GetType();
+			var key = (ownerType: type, propertyName);
+
+			if (_dependencyPropertyReflectionCache.TryGetValue(key, out var property))
+			{
+				return property;
+			}
+
+			property =
+				type
+					.GetTypeInfo()
+					.GetDeclaredProperty(propertyName)
+					?.GetValue(null) as DependencyProperty
+				?? type
+					.GetTypeInfo()
+					.GetDeclaredField(propertyName)
+					?.GetValue(null) as DependencyProperty;
+
+			_dependencyPropertyReflectionCache[key] = property;
```

Comment:
We do not validate the type of the value of the DP ... as it's now use for all types of control, I think it's a must.